### PR TITLE
Upload test result files to recipe artifact location

### PR DESCRIPTION
**Pull request by jmagman**
Copy test result files to recipe artifact location exposed in https://github.com/flutter/flutter/pull/74378.

`detailFiles` will include perf test trace files.

### CODE REVIEW

File: dev/devicelab/bin/run.dart [hunk 1]
```diff
@@ -139,6 +140,8 @@ Future<void> _runTasks() async {
       await cocoon.sendTaskResult(builderName: luciBuilder, result: result, gitBranch: gitBranch);
     }
 
+    hostAgent.dumpFiles(result.detailFiles);
```
Comment (CaseyHillers) [inline dev/devicelab/bin/run.dart:143]: Since this isn't dependent on flags, we should add this to `runTask` instead of `_runTasks`

**Review by CaseyHillers — APPROVED** [review]
LGTM

File: dev/devicelab/bin/run.dart (hunk 1 above)
Comment (jmagman) [reply dev/devicelab/bin/run.dart]: Thanks for the insight!

**Comment by jmagman** [comment]
```
00:07 +19 -1: test\host_agent_test.dart: dump files copies files and directories [E]                                                                                                                   
  Expected: true
    Actual: <false>
  
  package:test_api                 expect
  test\host_agent_test.dart 69:7   main.<fn>.<fn>
  test\host_agent_test.dart 53:42  main.<fn>.<fn>
```
Hmm Windows didn't like this.

**Comment by jmagman** [comment]
Closing in favor of `FLUTTER_TEST_OUTPUTS_DIR` recipe-side solution. https://flutter-review.googlesource.com/c/recipes/+/11622

**Pull request closed without merging**